src/Codec/Xlsx/SharedStringTable.hs

```diff
@@ -0,0 +1,167 @@
+
+{-------------------------------------------------------------------------------
+  Rendering
+-------------------------------------------------------------------------------}
+
+instance ToDocument SharedStringTable where
+  toDocument = documentFromElement "Shared string table generated by xlsx"
+             . toElement "sst"
+
+-- | See @CT_Sst@, p. 4452.
+instance ToElement SharedStringTable where
+  toElement nm SharedStringTable{..} = Element {
+      elementName       = nm
+    , elementAttributes = Map.empty
+    , elementNodes      = map (NodeElement . toElement "si")
+                        $ V.toList _sharedStringTable
+    }
+
+-- | See @CT_Rst@, p. 4443
+instance ToElement StringItem where
+  toElement nm si = Element {
+      elementName       = nm
+    , elementAttributes = Map.empty
+    , elementNodes      = map NodeElement $
+        case si of
+          StringItemText text -> [elementContent "t" text]
+          StringItemRich rich -> map (toElement "r") rich
+    }
+
+{-------------------------------------------------------------------------------
+  Parsing
+-------------------------------------------------------------------------------}
+
+-- | Parse shared string table
+--
+-- TODO: This is currently a VERY limited implementation, reusing the old
+-- parser for shared string tables. We lose all rich text information. Also,
+-- the translation from a list to an IntMap, then back to a list and finally
+-- to a vector is also unnecessary. We should just generate a list then and
+-- translate directly to a vector.
+sstParse :: Cursor -> SharedStringTable
+sstParse = aux . parseSharedStrings
+  where
+    aux :: IntMap Text -> SharedStringTable
+    aux = SharedStringTable . V.fromList . map StringItemText . IM.elems
+
+{-------------------------------------------------------------------------------
+  Extract shared strings
+-------------------------------------------------------------------------------}
+
+-- | Construct the 'SharedStringsTable' from an existing document
+sstConstruct :: [Worksheet] -> SharedStringTable
+sstConstruct =
+    SharedStringTable . V.fromList . uniq . concatMap goSheet
+  where
+    goSheet :: Worksheet -> [StringItem]
+    goSheet = mapMaybe (_cellValue >=> sstEntry) . Map.elems . _wsCells
+
+    sstEntry :: CellValue -> Maybe StringItem
+    sstEntry (CellText text) = Just $ StringItemText text
+    sstEntry (CellRich rich) = Just $ StringItemRich rich
+    sstEntry _               = Nothing
+
+    uniq :: Ord a => [a] -> [a]
+    uniq = Set.elems . Set.fromList
+
+sstLookupText :: SharedStringTable -> Text -> Int
+sstLookupText sst = sstLookup sst . StringItemText
+
+sstLookupRich :: SharedStringTable -> [RichTextRun] -> Int
+sstLookupRich sst = sstLookup sst . StringItemRich
+
+-- | Internal generalization used by 'sstLookupText' and 'sstLookupRich'
+--
+-- TODO: I don't know why this is using @(>=)@ instead of @(==)@ in the
+-- definition (behaviour inherited from the old sst search).
```

Comment:
searchFromTo is binary search algorithm by ordered data with complexity O(log n), so for correct work it requires predicate than return greather or equal condition